### DISCUSSION TITLE
Adaptive Stabilization Position - extruder just behind print

### DISCUSSION THREAD

**Discussion by LutzFassl** [FormerLurker/Octolapse#745] [Ideas]
Dear Formerlurker / Community!
Thanks for Octolapse. I'm having a lot of fun using it.

In the beginning I used the the stabilization positions that are completely away from the print. As you have also stated in the print quality troubleshooting guide, it's best to stay close to the print however (even if camera settings are already tuned for minimal snapshot time as well as retraction & temperature).
Therefore I switched back to centered stab. position. Which gives supergood print quality. However half of the print is covered by the extruder in the resulting video. So wouldn't it be nice to make a compromise between those?

I'm doing this manually now: I check the highest Y-coordinate of the print, (e.g. Ymax = 170mm) and set a custom stabilization position at Ymax + 20mm, e.g. Y190. And X I just keep centered (150mm in my case).

Could this be automated within octolapse? Meaning a stabilization position that is:
- fixed per print (as usual)

- but adaptive to each individual part
- at X=Centered and
- **Y = Ymax_of_print + user_defined_offset**

Maybe a feature like this exists already? Otherwise would be great to have it, I think.
Of course this would need to ignore that line that the printer draws in the beginning over the full Y-length, but this is most likely an easy hurdle to overcome.
BR, Lutz

PS: At least in case of Cura the MAXY value is already in the automatically generated above-header code:
![image](https://user-images.githubusercontent.com/37546799/123093862-79ad9280-d42c-11eb-9f55-64c389ab95a6.png)

**Comment by MrDB42** [comment]
I was just thinking about this today, as for a way to minimize travel distance.
